android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/home/views/HomeSurveysView.kt

```diff
@@ -0,0 +1,223 @@
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import co.nimblehq.ic.kmm.suv.android.R
+import co.nimblehq.ic.kmm.suv.android.extension.placeholder
+import co.nimblehq.ic.kmm.suv.android.ui.components.DotsIndicator
+import co.nimblehq.ic.kmm.suv.android.ui.theme.Typography
+import coil.compose.AsyncImage
+import timber.log.Timber
+
+data class HomeSurveysUiModel(
+    val title: String,
+    val description: String,
+    val imageUrlString: String,
+    val totalPages: Int,
+    val currentPageIndex: Int,
+    val isLoading: Boolean
+)
+
+@Composable
+fun HomeSurveysView(uiModel: HomeSurveysUiModel) {
+    if (uiModel.isLoading) {
+        HomeSurveysLoadingContent()
+    } else {
+        HomeSurveysContent(uiModel)
+    }
+}
+
+private enum class SwipeDirection {
+    IDLE, LEFT, RIGHT
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+private fun HomeSurveysContent(uiModel: HomeSurveysUiModel) {
+    val swipeableState = rememberSwipeableState(initialValue = SwipeDirection.IDLE)
+    val endAnchor = LocalDensity
+        .current
+        .run {
+            LocalConfiguration.current.screenWidthDp.toDp().toPx()
+        }
+    val anchors = mapOf(
+        endAnchor / 2 to SwipeDirection.IDLE,
+        0f to SwipeDirection.LEFT,
+        endAnchor to SwipeDirection.RIGHT
+    )
+
+    LaunchedEffect(swipeableState.currentValue) {
+        Timber.d("${swipeableState.currentValue}")
+        swipeableState.snapTo(SwipeDirection.IDLE)
+    }
+
+    Box(
+        modifier = Modifier
+            .fillMaxSize()
+            .swipeable(
+                state = swipeableState,
+                anchors = anchors,
+                orientation = Orientation.Horizontal
+            )
+    ) {
+        Box(modifier = Modifier.fillMaxSize()) {
+            AsyncImage(
+                model = uiModel.imageUrlString,
+                contentDescription = null,
+                contentScale = ContentScale.FillHeight,
+                modifier = Modifier.matchParentSize()
+            )
+        }
+        Box(
+            modifier = Modifier
+                .fillMaxSize()
+                .background(
+                    brush = Brush.verticalGradient(
+                        colors = listOf(
+                            Color.Transparent,
+                            Color.Black.copy(alpha = 0.6f)
+                        )
+                    )
+                )
+        ) {
+            Column(
+                verticalArrangement = Arrangement.Bottom,
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(horizontal = 20.dp)
+            ) {
+                DotsIndicator(uiModel.totalPages, uiModel.currentPageIndex)
+                Spacer(modifier = Modifier.height(20.dp))
```

Comment:
Some of the reused dimensions could be considered to define a common dimension for almost margin/padding ui aspects.